### PR TITLE
Introduce Pulumi - Modern Infrastructure as Code 🎉

### PR DESCRIPTION
**Pull request by ikovac**
**This PR**
- [x] Install Pulumi
- [x] Create custom VPC
- [x] Create an S3 bucket
- [x] Create a user with S3 access permission
- [x] Create Postgres instance
- [x] Create EC2 instance
- [x] Create and manage security groups for EC2 and Postgres
- [x] Generate RSA key pair for SSH access
- [x] Create an elastic IP address and associate it with the EC2 instance
- [x] Install the required software (Nginx, Nodejs, Certbot) and generate an SSL certificate for HTTPS connection
- [ ] Set up env variables after resources are deployed

**Steps to reproduce and test**
1. Sign in to the [Pulumi](https://www.pulumi.com/)
2. Do all `Get started` steps: https://www.pulumi.com/docs/get-started/aws/begin/ (Install Pulumi, configure AWS credentials...)
A good practice is to create a new non-root AWS user for Pulumi deployment with restricted access.
List of permissions which are required for this PR are listed below:
- AmazonRDSFullAccess
- AmazonEC2FullAccess
- IAMFullAccess
- AmazonS3FullAccess
3. Run `npm install` to install Pulumi packages
4. Navigate to the `pulumi/eip` folder
5. Run `pulumi stack init stack_name` for example `pulumi stack init dev`
6. Configure stack config. Run: `pulumi config set aws:region us-east-1`
7. Finally, create an elastic IP address by running `pulumi up` command.
As the result, the output should be something like this:
Screenshot 2020-12-10 at 11 05 17
where `ip` stands for the elastic IP address which is created for us.
8. Ping DNS admin and ask for test `extnsn.com` subdomain with `A record` set to the newly created elastic IP address.
9. Now, when we have our elastic IP address we can navigate to the `pulumi/infrastructure` folder
10. Init your stack (Same as step 5), run: `pulumi stack init dev`
11. For SSH access to the EC2 instance we will need to generate RSA keypair by running: `ssh-keygen -t rsa -f rsa -m PEM`. Tutorial: https://github.com/pulumi/examples/tree/master/aws-ts-ec2-provisioners
12. Change RSA private key permission to 400 by running: `chmod 400 rsa`
13. Set all config variables listed in `Pulumi.yaml` under the config section by running:
`pulumi config set CONFIG_NAME CONFIG_VALUE` and for secrets run: `pulumi config set CONFIG_NAME CONFIG_VALUE --secret`. For example, let's add RSA private and public keys to the pulumi config. 
- Public key: `cat rsa.pub | pulumi config set publicKey --`
- Private key: `cat rsa | pulumi config set privateKey --secret --`
14. Rename Nginx config file found in `scripts/ivo.extnsn.com` to your domain name
15. Run `pulumi up` 🚀  
16. SSH to your EC2 instance by running: `ssh -i privateKey ubuntu@DOMAIN_NAME`

❗❗ Feel free to ping me if you have any questions!

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -20374,6 +20665,11 @@
         "is-typedarray": "^1.0.0"
       }
     },
+    "typescript": {
+      "version": "3.7.5",
+      "resolved": "https://registry.npmjs.org/typescript/-/typescript-3.7.5.tgz",
+      "integrity": "sha512-/P5lkRXkWHNAbcJIiHPfRoKqyd7bsyCma1hZNUGfn20qm64T6ZBlrzprymeu918H+mB/0rIg2gGK/BXkhhYgBw=="
+    },
```
Comment (kronicker) [inline package-lock.json:20799]: @vladimyr 😱 😆

File: pulumi/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    Statement: [{
+      Sid: 'AllowRetouchS3Management',
+      Effect: 'Allow',
+      Resource: bucket.arn.apply(arn => arn + '/*'),
+      Action: 's3:*'
+    }]
+  }
+});
+
+// Attack policy to the user
```
Comment (vladimyr) [inline pulumi/index.js]: Attack? 😀

File: pulumi/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+// Create postgres db instance
+const db = new aws.rds.Instance('postgresdb', {
+  engine: 'postgres',
+  instanceClass: 'db.t2.micro',
+  allocatedStorage: 20,
+  name: 'tailor', // db name
+  username: 'tailor',
+  password: 'tailor123',
```
Comment (vladimyr) [inline pulumi/index.js]: Can we get this from dotenv?

Comment (ikovac) [reply pulumi/index.js]: Of course, this was the initial phase. It's not ready yet for review 😅 😄

Comment (vladimyr) [reply pulumi/index.js]: No worries I made sort of _todo comment_ there  😉
Blame @kronicker for my snooping 😀

**Comment by ikovac** [comment]
I will close this PR for now due to inactivity. Hope to reopen it soon 🤞

**Comment by ikovac** [comment]
I will close this PR for now due to inactivity. Hope to reopen it soon 🤞

**Pull request closed without merging**